### PR TITLE
feat: adds CAN budget form

### PR DESCRIPTION
**Pull request by fpigeonjr**
## What changed

- adds CAN budget form

## Issue

#310 #3099

## How to test

1. e2e tests pass
2. logging in as budget team
3. goto CANs details Funding Budget `/cans/504/funding`
4. select current fiscal year from drop down
5. enter edit mode via edit button
6. enter Budget Amount into input
7. click on `Add FY Budget` button
8. see value populate the card on the right `FY CAN Budget Card`
9. click on `Save Changes` button
10. see results on CAN Funding page

## Screenshots

_If relevant, e.g. for a front-end feature_

## Definition of Done Checklist
- [x] OESA: Code refactored for clarity
- [x] OESA: Dependency rules followed
- [x] Automated unit tests updated and passed
- [x] Automated integration tests updated and passed
- [x] Automated quality tests updated and passed
- [x] Automated load tests updated and passed
- [x] Automated a11y tests updated and passed
- [x] Automated security tests updated and passed
- [x] 90%+ Code coverage achieved
- [x] Form validations updated

### CODE REVIEW

File: frontend/src/pages/cans/detail/Can.jsx [hunk 1]
```diff
@@ -41,9 +42,16 @@ const Can = () => {
         budgetLineTypesCount,
         agreementTypesCount,
         receivedFunding,
-        isBudgetTeam
+        isBudgetTeam,
+        carryForwardFunding
     } = useCan();
 
+    const [isEditMode, setIsEditMode] = React.useState(false);
```
Comment (fpigeonjr) [inline frontend/src/pages/cans/detail/Can.jsx:49]: move logic to useCAN hook

File: frontend/src/pages/cans/detail/Can.hooks.js [hunk 2]
```diff
@@ -82,13 +88,14 @@ export default function useCan() {
...
         portfolioName: can?.portfolio?.name,
         portfolioId: can?.portfolio_id ?? -1,
-        totalFunding: CANFunding?.total_funding,
-        plannedFunding: CANFunding?.planned_funding,
-        obligatedFunding: CANFunding?.obligated_funding,
-        inExecutionFunding: CANFunding?.in_execution_funding,
-        inDraftFunding: CANFunding?.in_draft_funding,
-        expectedFunding: CANFunding?.expected_funding,
-        receivedFunding: CANFunding?.received_funding,
+        totalFunding: CANFunding?.total_funding ?? "0",
```
Comment (fpigeonjr) [inline frontend/src/pages/cans/detail/Can.hooks.js:91]: not sure on these fallback amounts, could 0 be actual budget amount?

**Review by rajohnson90 — APPROVED** [review]

**Review by maiyerlee — APPROVED** [review]

File: frontend/src/api/opsAPI.js [hunk 3]
```diff
@@ -212,6 +212,24 @@ export const opsApi = createApi({
...
+            invalidatesTags: ["Cans"]
+        }),
+        updateCanFundingBudget: builder.mutation({
+            query: ({ id, data }) => ({
+                url: `/can-funding-budgets/${id}`,
+                method: "PATCH",
+                headers: { "Content-Type": "application/json" },
+                body: data
+            }),
+            invalidatesTags: ["Cans"]
```
Comment (johndeange) [inline frontend/src/api/opsAPI.js:231]: I think you will want to add `CanFunding` tag to this list.  That way when you update the funding the `getCanFundingSummary` will re-fetch.

**Review by johndeange — APPROVED** [review]

File: frontend/src/api/opsAPI.js (hunk 3 above)
Comment (maiyerlee) [reply frontend/src/api/opsAPI.js:231]: Good catch!

**Pull request merged**